crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,301 @@
+            } else {
+                let error_string = response
+                    .error_code()
+                    .map(|error_code| error_code.to_string())
+                    .unwrap_or_else(|retcode| format!("Error with retcode: {}", retcode));
+                Err(AptosLedgerError::UnexpectedError(error_string))
+            }
+        },
+        Err(err) => Err(AptosLedgerError::UnexpectedError(err.to_string())),
+    }
+}
+
+/// Returns the official app name register in Ledger
+pub fn get_app_name() -> Result<String, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    match transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_GET_APP_NAME,
+        p1: 0,
+        p2: 0,
+        data: vec![],
+    }) {
+        Ok(response) => {
+            if response.retcode() == APDU_ANSWER_CODE {
+                let app_name = match str::from_utf8(response.data()) {
+                    Ok(v) => v,
+                    Err(e) => return Err(AptosLedgerError::UnexpectedError(e.to_string())),
+                };
+                Ok(app_name.to_string())
+            } else {
+                let error_string = response
+                    .error_code()
+                    .map(|error_code| error_code.to_string())
+                    .unwrap_or_else(|retcode| format!("Error with retcode: {}", retcode));
+                Err(AptosLedgerError::UnexpectedError(error_string))
+            }
+        },
+        Err(err) => Err(AptosLedgerError::UnexpectedError(err.to_string())),
+    }
+}
+
+/// Returns the public key of your Aptos account in Ledger device
+///
+/// # Arguments
+///
+/// * `display` - If true, the public key will be displayed on the Ledger device, and confirmation is needed
+pub fn get_public_key(display: bool) -> Result<String, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    // serialize the derivative path
+    let cdata = serialize_bip32(DERIVATIVE_PATH);
+
+    // APDU command's instruction parameter 1 or p1
+    let p1: u8 = match display {
+        true => 0x01,
+        false => 0x00,
+    };
+
+    match transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_GET_PUB_KEY,
+        p1,
+        p2: 0,
+        data: cdata,
+    }) {
+        Ok(response) => {
+            // Got the response from ledger after user has confirmed on the ledger wallet
+            if response.retcode() == APDU_ANSWER_CODE {
+                // extract the Public key from the response data
+                let mut offset = 0;
+                let response_buffer = response.data();
+                let pub_key_len: usize = (response_buffer[offset] - 1).into();
+                offset += 1;
```

Comment:
`offset += 1` is duplicated

Reply:
Its actually intentional :) 

Reference from the Petra Ledger side - https://github.com/aptos-labs/wallet/blob/main/apps/extension/src/core/ledger/index.ts#L163